openshift/insights-operator
# Insights recommendation metric and alert

Same as #630, except directly into 4.12 instead of `master`.

## Manual testing steps

1. Launch a cluster-bot cluster on this PR.
2. Copy and run the `oc login` command for said cluster.
3. Run the following commands.
```sh
oc scale --namespace=openshift-cluster-version deployment/cluster-version-operator --replicas 0
oc new-project project1
oc new-project project2
oc patch netnamespace project1 --type=merge -p '{"egressIPs":["192.168.1.99"]}'
oc patch netnamespace project2 --type=merge -p '{"egressIPs":["192.168.1.99"]}'
oc patch storage.operator.openshift.io cluster --type merge --patch '{"spec":{"managementState": "Unmanaged"}}'
oc create secret generic support --namespace=openshift-config --from-literal=endpoint=https://cloud.redhat.com/api/ingress/v1/upload --from-literal=interval=10m
oc scale -n openshift-insights deployment/insights-operator --replicas 0
oc scale -n openshift-insights deployment/insights-operator --replicas 1
```
4. Check the Observe > Alerting tab in OpenShift Web Console. You should see 3 instances of `InsightsRecommendationActive` alert there (they will be pending for the first 5 minutes).

## Categories



## Sample Archive


N/A

## Documentation


N/A

## Unit Tests


N/A

## Privacy


Yes. There are no sensitive data in the newly collected information.

## Changelog


No.

## Breaking Changes


No.

## References


Jira Task: https://issues.redhat.com/browse/CCXDEV-7327
